### ISSUE TITLE
Problem with mAP values

### ISSUE THREAD

**Issue by Hieung28** [AlexeyAB/darknet#4668]
Hello everyone,
I have been training YoloV3 network with my own dataset following the instruction of @AlexeyAB.
but i got some problems with mAP values. As you can see in the figure in my attach files, the mAP chart is quite terrible with much vibration and the maximum value is extremely lower than expected.
The attach files are image of loss/mAP chart, a labeled objects file, my yolo configure file and my command script file to start training 

I also trained with the Coco dataset instead  but the result was no more better.
Could you please give me some suggestions !!
Thank you
![chart](https://user-images.githubusercontent.com/59725606/72174865-42e8e580-340d-11ea-84b9-c6aaf307dab6.png)
[Yolo pnid cfg.txt](https://github.com/AlexeyAB/darknet/files/4046990/Yolo.pnid.cfg.txt)
[D-200-1225-1001_5_0010-1_0_2_480.txt](https://github.com/AlexeyAB/darknet/files/4046991/D-200-1225-1001_5_0010-1_0_2_480.txt)
[Training Cmd.txt](https://github.com/AlexeyAB/darknet/files/4046997/Training.Cmd.txt)

**Comment by AlexeyAB** [comment]
Run training command with `-show_imgs` flag at the end.
And show screenshot of your training images with bounded boxes.

**Comment by Hieung28** [comment]
> Run training command with `-show_imgs` flag at the end.
> And show screenshot of your training images with bounded boxes.

![aug_1181748379_0_-1411203788](https://user-images.githubusercontent.com/59725606/72197273-10ada700-3452-11ea-8b7b-70220bc82d97.jpg)
![aug_1313999695_0_-1438798280](https://user-images.githubusercontent.com/59725606/72197275-10ada700-3452-11ea-9a93-6546fc77b414.jpg)
![Label Review 1](https://user-images.githubusercontent.com/59725606/72197281-21f6b380-3452-11ea-8c78-ba4bffbe1755.PNG)
![Label Review 2](https://user-images.githubusercontent.com/59725606/72197282-21f6b380-3452-11ea-9c82-a57cca9f2518.PNG)
The first two images are screenshot with bounded boxes by running command with -show_imgs flag at the end
The last two images are screenshot with bounded boxes by my own tool. I did make a tool that check bounded boxes after labeling objects

**Comment by AlexeyAB** [comment]
> The first two images are screenshot with bounded boxes by running command with -show_imgs flag at the end

But I don't see bboxes. It means that your training dataset is incorrect.

> The last two images are screenshot with bounded boxes by my own tool. I did make a tool that check bounded boxes after labeling objects

Is it blue bboxes?

**Comment by Hieung28** [comment]
> > The first two images are screenshot with bounded boxes by running command with -show_imgs flag at the end
> 
> But I don't see bboxes. It means that your training dataset is incorrect.
> If you check carefully, you will see a blurred bounded box
> > The last two images are screenshot with bounded boxes by my own tool. I did make a tool that check bounded boxes after labeling objects
> 
> Is it blue bboxes?
yes, blue bboxes

**Comment by Hieung28** [comment]
> > The first two images are screenshot with bounded boxes by running command with -show_imgs flag at the end
> 
> But I don't see bboxes. It means that your training dataset is incorrect.
> 
> > The last two images are screenshot with bounded boxes by my own tool. I did make a tool that check bounded boxes after labeling objects
> 
> Is it blue bboxes?

**Comment by thelightsky** [comment]
Hi @AlexeyAB i have got same issue.
I guest the object in the picture is very small with the picture area.
Additionally, the anchor box in the cfg file are very different with truth box.

Have you any comments/questions for this one .
Thanks.

**Comment by Hieung28** [comment]
> Hi @AlexeyAB i have got same issue.
> I guest the object in the picture is very small with the picture area.
> Additionally, the anchor box in the cfg file are very different with truth box.
> 
> Have you any comments/questions for this one .
> Thank
Have you solved your problem?

**Comment by Hieung28** [comment]
@thelightsky  Have you solved your problem?

**Comment by Hieung28** [comment]
@AlexeyAB I have been stuck in this issue for a long time, do you have any suggestions or comments?

**Comment by AlexeyAB** [comment]
> If you check carefully, you will see a blurred bounded box

* I don't see it. Can you show them with arrows?

* Also set valid=train.txt in obj.data file - what mAP can you get?

* How many classes are in training dataset, and how many images are in training dataset?

Read: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection
**for each object which you want to detect - there must be at least 1 similar object in the Training dataset with about the same: shape, side of object, relative size, angle of rotation, tilt, illumination.** So desirable that your training dataset include images with objects at diffrent: scales, rotations, lightings, from different sides, on different backgrounds - you should preferably have 2000 different images for each class or more, and you should train 2000*classes iterations or more

**Comment by Hieung28** [comment]
![Inked72197273-10ada700-3452-11ea-8b7b-70220bc82d97_LI](https://user-images.githubusercontent.com/59725606/72457070-625e8480-37f8-11ea-9ac6-eecf5e3dd1af.jpg)
@AlexeyAB  Here is the blurred box in training image

**Comment by Hieung28** [comment]
![chart](https://user-images.githubusercontent.com/59725606/72457147-8e7a0580-37f8-11ea-9ddc-1997f16e6f4f.png)
@AlexeyAB  mAP value i got after setting valid-train.txt  in obj.data file

**Comment by AlexeyAB** [comment]
@Hieung28 If you get high mAP on train set and low on val-set, then your train set isn't representative.

Read: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection
for each object which you want to detect - there must be at least 1 similar object in the Training dataset with about the same: shape, side of object, relative size, angle of rotation, tilt, illumination. So desirable that your training dataset include images with objects at diffrent: scales, rotations, lightings, from different sides, on different backgrounds - you should preferably have 2000 different images for each class or more, and you should train 2000*classes iterations or more

**Comment by Hieung28** [comment]
@AlexeyAB The result after 6000 literations. Do you know why mAP is not stable?
![chart](https://user-images.githubusercontent.com/59725606/72484407-b9834a00-3836-11ea-862c-38c76c47a814.png)

**Comment by AlexeyAB** [comment]
Low batch-size, small training and validation datasets....

**Comment by Hieung28** [comment]
@AlexeyAB Hey sir, Is reasonable if i train with datasets like the following image
![image](https://user-images.githubusercontent.com/59725606/74229974-05ba9080-4cf6-11ea-9fd0-10b3231a8506.png)
